### DISCUSSION TITLE
Resource control on blobfuse

### DISCUSSION THREAD

**Discussion by songjiaxun** [kubernetes-sigs/blob-csi-driver#768] [Ideas]
I have a question about the resource control on blobfuse when the proxy mode is being used.

Since the blobfuse binary runs directly on the nodes, if there is no resource control on the blobfuse instances, when large files / big amount of files are being processed, the blobfuse instance may consume substantial cpu/memo resources. And if the resource usage is not accounted, there might be a mismatch between allocatable cpu/memo in kubelet and the real state.

Does Azure blobfuse CSI have any plan to implement resource control? Thank you!

**Comment by andyzhangx** [comment]
@songjiaxun we don't have any plan to do this. The proxy is out of k8s control, not sure how to do that. Do you have similar requirement on GCP?

**Reply by songjiaxun** [reply]
Yes, we are developing GCS CSI Driver on GKE. Because we will support the driver on [GKE Autopilot](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview) clusters, and the user don't pay for the node system resources, we are considering how to limit the fuse resource usage on nodes.

We have explored multiple options, including moving around the fuse process cgroup to make it accounted under a Pod/container cgroup. But we could not figure out a good solution.

So I am curious about how AKS handles the resource usage control and monitoring. If AKS does not do it, do users complain about their node allocatable resources being lower than what they meant to be when they use blobfuse to transfer large files? For example, I see blobfuse could consume a lot of memory in this [ticket](https://github.com/Azure/azure-storage-fuse/issues/454).

**Reply by andyzhangx** [reply]
that's a good question, I think that's by design for blobfuse, we should mention that in CSI driver repo.
